include/grpcpp/security/tls_certificate_provider.h

```diff
@@ -36,6 +40,11 @@ class GRPCXX_DLL CertificateProviderInterface {
  public:
   virtual ~CertificateProviderInterface() = default;
   virtual grpc_tls_certificate_provider* c_provider() = 0;
+
+  // Returns an OK status if the credentials held by the provider are valid.
+  // What it means for a credential to be valid is determined by the provider
+  // implementation.
+  virtual absl::Status ValidateCredentials() const { return absl::OkStatus(); }
```

Comment:
I don't think it makes sense for this to be a method on the `CertificateProviderInterface`, because the interface is actually designed for more general-purpose cases where there can be multiple certs provided by the same provider, and this API is ambiguous in cases where some of the certs are valid and others are not.

I think this method makes sense only for provider implementations that supply only a single cert, such as the file-watcher and static providers.  Given that, I think we should support this method only on the file-watcher and static providers specifically.  Applications will know which specific subclass they instantiated, so it should be fine for them to call this method.

Reply:
That sounds reasonable, thanks. Removed from the base class.